### ISSUE TITLE
publish_as_nanopub() broken by requirement to publish all steps individually

### ISSUE THREAD

**Issue by raar1** [fair-workflows/fairworkflows#165]
```publish_as_nanopub()``` as used in the ```noodles_fw.ipynb``` example notebook is currently broken, complaining ``` [...] was not published yet, please publish steps first```. Should this not be the default behaviour? If user wishes to publish the workflow they have just constructed, I think it should by default publish each of the steps too. Or is there another way this is intended to be used?

**Comment by svenvanderburg** [comment]
Tobias and Remzi wanted this behavior. I can see why you would disagree though. Nevertheless I would vote for just leaving it like it is now, otherwise we are meandering between different behaviors all the time. @raar1 do you agree to close the issue?

**Comment by svenvanderburg** [comment]
Me & @raar1 decided we should have a non-default option to publish the steps, something like this:
```python
def publish_as_nanopub(publish_steps=False):
    if publish_all:
        for step in workflow:
            step.publish_as_nanopub()
```

**Comment by svenvanderburg** [comment]
Fixed by #179
